components/button.tsx

```diff
@@ -2,15 +2,24 @@ import Link from "next/link"
 
 interface ButtonProps {
   children: string
-  to: string
+  href?: string
+  type?: "button" | "reset" | "submit"
 }
 
-export default function Button ({ children, to }: ButtonProps): JSX.Element {
-  return (
-    <Link href={to}>
-      <button className="govuk-button lbh-button">
-        {children}
-      </button>
-    </Link>
+export default function Button({ children, href, type }: ButtonProps): JSX.Element {
+  const button = (
+    <button className="govuk-button lbh-button" type={type}>
+      {children}
+    </button>
   )
+
+  if (href) {
+    return (
+      <Link href={href}>
```

Comment:
This feels like an odd use of button, since it will only ever act as an `a` tag and not a button?

Reply:
Unfortunately, you cannot wrap the custom `<Button />` component with `<Link />`, for some reason that did not work - it's like it expects the component beneath it to be a button or anchor tag, so I had to move the link functionality into the button component.

I think this is fine, because in terms of display it appears as a button